lib/fs.js

```diff
@@ -1794,10 +1807,17 @@ function WriteStream(path, options) {
     options = {};
   else if (typeof options === 'string')
     options = { encoding: options };
-  else if (options === null || typeof options !== 'object')
-    throw new TypeError('options must be a string or an object');
-
-  options = Object.create(options);
+  else if (options === null || typeof options !== 'object') {
+    //throw new TypeError('options must be a string or an object');
+    warnedWriteStreamOptions = internalUtil.printDeprecationMessage(
+      'Passing anything but an object or a string as the second argument to \
+WriteStream is deprecated.',
+      warnedWriteStreamOptions
+    );
+    options = options || {};
+  } else {
```

Comment:
Sorry, I meant, `options = Object.create(options);` used to be outside the `if..else if` blocks in both the cases. Now it is only inside the `else` part. Isn't this wrong?